differential-dataflow/src/operators/enter.rs

```diff
@@ -0,0 +1,54 @@
+//! Enter a collection into a scope.
+
+use timely::Data;
+use timely::dataflow::{Scope, ScopeParent, Stream, StreamCore};
+use timely::dataflow::operators::core::{Enter, Map};
+use timely::dataflow::scopes::Child;
+use timely::progress::timestamp::Refines;
+
+/// Extension trait for streams.
+pub trait EnterTime<C, G, TInner>
```

Comment:
As with the `Negate` comment, I think `G, TInner, C` is the more common order; certainly the scope comes first in just about all timely traits iirc (both `Map` and `Operator`, that I just checked). Timely's `Enter` trait is `Enter<G, T, C>` fwiw.